### ISSUE TITLE
DL3026 fails if using an ARG

### ISSUE THREAD

**Issue by Fleshgrinder** [hadolint/hadolint#315]
- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

I can whitelist `ARG`s that are allowed in `FROM`.

### Actual behavior

Fails with `Use only an allowed registry in the FROM image` despite `--trusted-registry '${MY_ARG}'` being set.

### Steps to reproduce the behavior

```Dockerfile
# Dockerfile
ARG MY_BASE
FROM ${MY_BASE}/image:tag
```

```shell
docker run --rm hadolint/hadolint hadolint --trusted-registry '${MY_BASE}' Dockerfile
```

### Relates to

* #219 
* #239

**Comment by Fleshgrinder** [comment]
Just realized that this is actually not possible with Docker. 🤣 Seems like the `FROM ARG` is not intelligent enough to allow you changing the registry with an `ARG`. I'll first make sure that Docker can do that before complaining here. Hence, I might reopen this in the future.

**Comment by lucas85cunha** [comment]
Did you find the solution? Because I try use the hadolint on my pipeline and recive the same error.
